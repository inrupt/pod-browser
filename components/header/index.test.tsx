import * as ReactFns from "react";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";

import Header from "./index";

describe("Header", () => {
  describe("with user logged in", () => {
    test("renders with a logout button", () => {
      jest.spyOn(ReactFns, "useContext").mockImplementation(() => ({
        session: true,
      }));

      const tree = shallow(<Header />);
      expect(shallowToJson(tree)).toMatchSnapshot();
    });
  });

  describe("with user logged out", () => {
    test("renders without a logout button", () => {
      jest.spyOn(ReactFns, "useContext").mockImplementation(() => ({
        session: undefined,
      }));

      const tree = shallow(<Header />);
      expect(shallowToJson(tree)).toMatchSnapshot();
    });
  });
});
