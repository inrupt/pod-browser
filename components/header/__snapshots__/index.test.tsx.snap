// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header with user logged in renders with a logout button 1`] = `
<nav>
  <WithStyles(ForwardRef(Link))>
    Inrupt
  </WithStyles(ForwardRef(Link))>
  <WithStyles(ForwardRef(Link))>
    <i
      className="prism-icon-add"
    />
  </WithStyles(ForwardRef(Link))>
  <LogOut />
</nav>
`;

exports[`Header with user logged out renders without a logout button 1`] = `
<nav>
  <WithStyles(ForwardRef(Link))>
    Inrupt
  </WithStyles(ForwardRef(Link))>
  <WithStyles(ForwardRef(Link))>
    <i
      className="prism-icon-add"
    />
  </WithStyles(ForwardRef(Link))>
</nav>
`;
